src/commands/up.rs

```diff
@@ -646,6 +667,87 @@ fn trigger_command_for_resolved_app_source(
         .collect()
 }
 
+// Scrubs the locked app to only contain the given list of components
+// Introspects the LockedApp to find and selectively retain the triggers that correspond to those components
+fn retain_components(locked_app: &mut LockedApp, components: &[String]) -> Result<()> {
```

Comment:
Am I right in remembering that Wasm files have to exist in order to build a LockedApp?  That is, suppose I did a `spin up` where I excluded `component-xyz`: would it still require me to have `component_xyz.wasm`?

Reply:
I am pretty sure the wasm files do not have to exist; rather it builds the lockfile based on the manifest and where it says the file should be. For example, in this test that passes, the wasm file doesn't exist: https://github.com/kate-goldenring/spin/blob/67af2d3e2fbdce7c9113fa3c0fa74e4ff8151ee2/src/commands/up.rs#L778-L779